packages/@lwc/engine/src/framework/api.ts

```diff
@@ -140,18 +132,14 @@ const ElementHook: Hooks<VElement> = {
         insertNodeHook(vnode, parentNode, referenceNode);
         createChildrenHook(vnode);
     },
-    move: (vnode, parentNode, referenceNode) => {
-        insertNodeHook(vnode, parentNode, referenceNode);
-    },
-    remove: (vnode, parentNode) => {
-        removeNodeHook(vnode, parentNode);
-        removeElmHook(vnode);
-    },
+    move: insertNodeHook,
+    remove: removeNodeHook,
 };
 
 const CustomElementHook: Hooks<VCustomElement> = {
     create: vnode => {
         const { sel } = vnode;
+        registerTagName(sel);
```

Comment:
lets make sure that this is the most optimal place to do this operation. I don't think it is, but we will see.

Reply:
Are you thinking about how to invoke registerTagName() only once for a given tagName?

I looked into this a bit, the only common shared data between two elements of the same tagName is their class constructor. The constructor is processed only once. But unfortunately, we have decoupled the constructor from the tag name. So we cannot register the tagName when we process the constructor.

Reply:
I mean, for now, it is fine... I mean in the future. Maybe the local registry will solve this problem.